### ISSUE TITLE
FR: Expression language support for removing query params from request.path attribute

### ISSUE THREAD

**Issue by douglas-reid** [istio/istio#6380]
**Is your feature request related to a problem? Please describe.**
There is a desire to support telemetry related to http endpoints. Right now, the API Mgmt support does not exist in Istio to allow mapping to `api.operation` via OpenAPI. A method for trimming down `request.path` to just the essentials is needed to fill the void.

**Describe the solution you'd like**
Add a func like: `urlPath(string)` to the expression language. It should function as follows:
```
endpoint: urlPath(request.path) | "unknown"
```

Given: `request.path = "/foo/bar?x=10&mesh=istio"`, it should produce `"/foo/bar"`.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by douglas-reid** [comment]
cc: @geeknoid

**Comment by geeknoid** [comment]
We can add a regex capable replace function that takes a regex as input and
a replacement pattern as output. Should be only a few lines of glue code to
add this and call out to the Go regex logic.

**Comment by kyessenov** [comment]
Alternatively: this is derivable in the proxy. We could supply `request.stripped_path` to be the query-stripped path (that is actually what envoy matches against, not `:path` pseudo-header) and a query string map `request.path_query`.

**Comment by sakshigoel12** [comment]
@douglas-reid can you provide an ETA for this issue?

**Comment by bianpengyuan** [comment]
@sakshigoel12 I am going to work on this and ETA is next week.

**Comment by huydinhle** [comment]
@bianpengyuan  I see the https://github.com/istio/proxy/pull/1837 got merged into proxy and released with 1.0.2. I tried 1.0.2 out but I still don't see the `url_path` in our logentry. can you help me to debug this. 

```
{"level":"info","time":"2018-10-08T20:57:42.722879Z","instance":"accesslog.logentry.istio-next","apiClaims":"","apiKey":"","clientTraceId":"","connection_security_policy":"none","destinationApp":"api-flight","destinationIp":"10.240.33.110","destinationName":"api-flight-8454979484-pft9j","destinationNamespace":"legacy","destinationOwner":"kubernetes://apis/extensions/v1beta1/namespaces/legacy/deployments/api-flight","destinationPrincipal":"","destinationServiceHost":"api-flight.legacy.svc.cluster.local","destinationWorkload":"api-flight","httpAuthority":"10.240.33.110:8080","latency":"4.464568ms","method":"GET","protocol":"http","receivedBytes":231,"referer":"","reporter":"destination","requestId":"c13a70b7-f090-9e48-8878-23976ef36fb6","requestSize":0,"requestedServerName":"","responseCode":200,"responseSize":0,"responseTimestamp":"2018-10-08T20:57:42.727263Z","sentBytes":279,"sourceApp":"","sourceIp":"0.0.0.0","sourceName":"unknown","sourceNamespace":"default","sourceOwner":"unknown","sourcePrincipal":"","sourceWorkload":"unknown","url":"/health","urlPath":"","userAgent":"kube-probe/1.10","xForwardedFor":"0.0.0.0"}
```

**Comment by huydinhle** [comment]
I also see this https://github.com/istio/api/pull/566/files, so I rebuilt mixer with this changes pulled to test it out but still the labels `url_path` does not show up at all

**Comment by huydinhle** [comment]
@bianpengyuan  I added this 
```
    urlPath: request.url_path | ""
```
into that accesslog logentry file but the urlPath comes up as empty string `""`

**Comment by huydinhle** [comment]
@bianpengyuan ping ! 

Any update?

**Comment by bianpengyuan** [comment]
@huydinhle Hey, I checked again, actually the feature is not implemented in 1.0 branch. It is in master branch and should be ready in 1.1 release.

**Comment by huydinhle** [comment]
@bianpengyuan I saw the code for this feature in these 2 PRs.
 - https://github.com/istio/proxy/pull/1837 --> this is in istio/proxy:1.0.2
 - https://github.com/istio/api/pull/566 --> this is only in master of istio/istio

So I build a mixer image with the second PR( https://github.com/istio/api/pull/566) and rolled it into our istio setup. But the urlPath is still missing. 

Am I missing anything to get this feature to work?

**Comment by bianpengyuan** [comment]
@huydinhle I don't think istio/proxy#1837 is merged into 1.0 branch. I am wondering why do you think so?

**Comment by huydinhle** [comment]
ouch my bad, I thought these are the commits that went into 1.0.2 https://github.com/istio/proxy/compare/1.0.2...master. My mistake, sorry about that @bianpengyuan

**Comment by smawson** [comment]
This was fixed in 1.1 right? Closing assuming so.
